### ISSUE TITLE
Add offers to sitemap

### ISSUE THREAD

**Issue by timoludwig** [Integreat/cms-django#493]
### Motivation

At the moment, only pages, events and pois are included in the sitemap. Offers are missing.

### Proposed Solution

Add all routes to offers to the sitemap. Create a new class `OfferSitemap` in [sitemaps.py](https://github.com/Integreat/cms-django/blob/develop/src/sitemap/sitemaps.py) which inherits from `WebappSitemap` similar to e.g. `PageSitemap`.

The queryset should be:
```
queryset = Offer.objects.all()
```
which should then be filtered in the `__init__()`-method as follows:
```
self.queryset = self.queryset.filter(
    region=self.region
)
```
Then, add the function `get_absolute_url()` to the [Offer model](https://github.com/Integreat/cms-django/blob/develop/src/cms/models/offers/offer.py) which should return the absolute url in the following form:
`<region_slug>/<language_code>/offers/<offer_slug>`

In the end, add `OfferSitemap` to the `sitemaps` list in [src/sitemap/utils.py:27](https://github.com/Integreat/cms-django/blob/d905ab25d8425c829972e70da1e3b9f25ea61536/src/sitemap/utils.py#L27).

Please also add docstrings to all classes and methods.

**Comment by stanislavkocan** [comment]
Hello, I would like to contribute and solve this issue. Can you please assign it to me ?

**Comment by timoludwig** [comment]
Hey @stanislavkocan, cool! :rocket:  
I updated the issue description to give more details on the proposed solution.

**Comment by stanislavkocan** [comment]
Thanks @timoludwig :) I would like to ask you, what should be the language_code ? If I understand correctly, offers are same in English and German version, that's why we filter only by region. It means that same offers should be in both sitemaps just with different language_code ? 

Right now, I set language_code to `self.region.default_language.code` (that was the only way how I knew to get language), but offers appear only in "German" sitemap with code _de-de_.

Thank you :)

**Comment by timoludwig** [comment]
Ah sorry, I forgot about this problem. Yes, the offers are the same for different languages. But we want to include them in all sitemaps for all active languages. One example:

- https://integreat.app/augsburg/de/offers/sprungbrett
- https://integreat.app/augsburg/en/offers/sprungbrett

are both identical, but we need both urls in the sitemap. Clearly we can't access the required language parameter from the model, so my approach with implementing `get_absolute_url()` won't work. My next idea would be to add a function `location()` to the `OfferSitemap` class... if I understand [the docs](https://docs.djangoproject.com/en/2.2/ref/contrib/sitemaps/#django.contrib.sitemaps.Sitemap.location) correctly, this is used to retrieve the urls of items before the fallback to `get_absolute_url()`.

**Comment by stanislavkocan** [comment]
Ok, I will take a look. 

However, does it make sense to filter it by region ? 

`self.queryset = self.queryset.filter(
    region=self.region
)`

I deleted it and right now offers appear in english version too, just with language_code _de-de_. Now I only need to somehow implement this location you mentioned. 

Correct me if I am wrong, my first issue.

Thanks

**Comment by timoludwig** [comment]
> However, does it make sense to filter it by region ?

Yes, filtering by region is necessary, because offers should only appear in the region's sitemap if they are activated for that specific region. I didn't test the syntax, so maybe my suggestion doesn't work as expected, but filtering is indeed necessary. You can also give `self.queryset = self.region.offers` a try.

> I deleted it and right now offers appear in english version too, just with language_code _de-de_. Now I only need to somehow implement this location you mentioned.

I'm not sure why they appear when you delete the filter, maybe you activated the offer for another region than you are requesting the sitemap? You can also push your changes to a feature branch of your fork so I can have a look at your progress if you want.

**Comment by stanislavkocan** [comment]
Yes, now it's clear to me. It seems like it works with _location(),_ I will do some testing and then push.

**Comment by timoludwig** [comment]
Fixed by @stanislavkocan in #511
